src/main/kotlin/org/ozwillo/dcimporter/service/MarcheSecuriseListingService.kt

```diff
@@ -0,0 +1,253 @@
+package org.ozwillo.dcimporter.service
+
+import org.ozwillo.dcimporter.config.DatacoreProperties
+import org.ozwillo.dcimporter.model.BusinessAppConfiguration
+import org.ozwillo.dcimporter.model.BusinessMapping
+import org.ozwillo.dcimporter.model.datacore.DCOperator
+import org.ozwillo.dcimporter.model.datacore.DCOrdering
+import org.ozwillo.dcimporter.model.datacore.DCQueryParameters
+import org.ozwillo.dcimporter.model.marchepublic.*
+import org.ozwillo.dcimporter.model.sirene.Organization
+import org.ozwillo.dcimporter.repository.BusinessAppConfigurationRepository
+import org.ozwillo.dcimporter.repository.BusinessMappingRepository
+import org.ozwillo.dcimporter.util.*
+import org.ozwillo.dcimporter.util.soap.response.parsing.ResponseObject
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import org.springframework.web.client.HttpClientErrorException
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.core.publisher.toFlux
+import reactor.core.publisher.toMono
+import java.lang.Exception
+import java.lang.IllegalArgumentException
+import java.time.Duration
+import java.time.Instant
+
+@Service
+class MarcheSecuriseListingService(private val datacoreService: DatacoreService,
+                                   private val datacoreProperties: DatacoreProperties,
+                                   private val businessMappingRepository: BusinessMappingRepository,
+                                   private val businessAppConfigurationRepository: BusinessAppConfigurationRepository){
+
+    companion object {
+        private val logger: Logger = LoggerFactory.getLogger(MarcheSecuriseListingService::class.java)
+    }
+
+    private val startTime = Instant.now()
+    private var triggeredError = 0
+
+    private val MP_PROJECT = "marchepublic_0"
+
+    @Value("\${datacore.model.modelORG}")
+    private val modelOrg = ""
+
+    @Value("\${marchesecurise.url.updateConsultation}")
+    private val updateConsultationUrl = ""
+    @Value("\${marchesecurise.url.registre}")
+    private val registreUrl = ""
+
+    fun refreshDatacoreRegistreReponse(){
+        val ordre = Ordre.DATE_PREMIER_RETRAIT.toString()
+        val sensOrdre = SensOrdre.ASC.toString()
+        val startParam = 0
+        val limitParam = Int.MAX_VALUE
+
+        logger.debug("Beginning Datacore register update from Marchés Sécurisés ...")
+
+        listOf(MSUtils.REPONSE_TYPE, MSUtils.RETRAIT_TYPE)
```

Comment:
there is far too much business code in this function (and is hard to read and understand), split it into meaningful and simpler functions.

Reply:
A new cleaned version is available in October 16th commit. Let me now if more have to be done.